### PR TITLE
feat(geometry): implement closed-form solve_quartic

### PR DESCRIPTION
**Pull request by zeel991**
## 📝 Description

**⚠️ Issue Link Required**: Fixes # (Insert your Issue Number here)

**Fixes :** https://github.com/kornia/kornia/issues/3532

**Important**:
- Ensure you are assigned to the linked issue before submitting this PR
- This PR should strictly implement what the linked issue describes
- Do not include changes beyond the scope of the linked issue

---

## 🛠️ Changes Made
- [x] Implemented `solve_quartic` in `kornia/geometry/solvers/polynomial_solver.py` using a differentiable, vectorized version of **Ferrari's Method**.
- [x] Added numerical stability safeguards (e.g., `torch.clamp` / `torch.relu` before square roots) to prevent exploding gradients when discriminants are near zero.
- [x] Implemented graceful fallback to `solve_cubic` when the leading coefficient ($a=0$) is zero.
- [x] Added comprehensive test suite `TestQuarticSolver` in `tests/geometry/solvers/test_polynomial_solver.py`.

---

## 🧪 How Was This Tested?
- [x] **Unit Tests:** Added `TestQuarticSolver` covering:
    - `test_smoke`: Verifies roots for known polynomials.
    - `test_random`: Generates random roots, constructs polynomials via Vieta's formulas, and verifies residuals are $\approx 0$.
    - `test_solve_quartic`: Tests specific edge cases like double roots and fallback to cubic.
    - `gradcheck`: Verified differentiability using `torch.autograd.gradcheck` on `float64` tensors.
- [x] **Manual Verification:** Ran local test suite with `pytest tests/geometry/solvers/test_polynomial_solver.py` (all tests passed).
- [x] **Performance/Edge Cases:** Validated batch processing support (shapes B=1, 2, 4, 7) and stability for coefficients where $a=0$.

---

## 🕵️ AI Usage Disclosure
*Check one of the following:*
- [ ] 🟢 **No AI used.**
- [x] 🟡 **AI-assisted:** I used AI for boilerplate/refactoring but have manually reviewed and tested every line.
- [ ] 🔴 **AI-generated:** (Note: These PRs may be subject to stricter scrutiny or immediate closure if the logic is not explained).

---

## 🚦 Checklist
- [x] I am assigned to the linked issue (required before PR submission)
- [x] The linked issue has been approved by a maintainer
- [x] I have performed a **self-review** of my code (no "ghost" variables or hallucinations).
- [x] My code follows the existing style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] (Optional) I have attached screenshots/recordings for UI changes.

---

## 💭 Additional Context
The implementation follows the requested analytical approach (Ferrari's method) rather than an iterative one, ensuring determinism and easier gradient propagation. I verified the gradients using double precision to ensure stability.

### CODE REVIEW

File: kornia/geometry/solvers/polynomial_solver.py [hunk 1]
```diff
@@ -213,9 +213,226 @@ def solve_cubic(coeffs: torch.Tensor) -> torch.Tensor:
...
+    q2_b = 0.5 * A + R
+    q2_c = 0.5 * y + E
+
+    roots1 = solve_quadratic(torch.stack([q1_a, q1_b, q1_c], dim=1))
+    roots2 = solve_quadratic(torch.stack([q2_a, q2_b, q2_c], dim=1))
+
+    solutions[mask_quartic, 0:2] = roots1
+    solutions[mask_quartic, 2:4] = roots2
+
+    return solutions
```
Comment (sidd-27) [inline kornia/geometry/solvers/polynomial_solver.py]: ?? why is there code being repeated

Comment (zeel991) [reply kornia/geometry/solvers/polynomial_solver.py]: Apologies for the confusion, The duplication happened while resolving a merge conflict during the rebase, I have cleaned up the file and pushed the fix. It should be correct now.

**Comment by zeel991** [comment]
> /review

I have addressed all the feedback from the review
